### PR TITLE
Fix daemon missing group

### LINKED ISSUE #456: multipassd fails when sudo group does not exist
I've installed multipass on Arch (either using --beta or --edge), multipassd exited with error:

```
maciek@galeon:/snap/multipass/current systemctl status snap.multipass.multipassd.service
● snap.multipass.multipassd.service - Service for snap application multipass.multipassd
   Loaded: loaded (/etc/systemd/system/snap.multipass.multipassd.service; enabled; vendor preset: disabled)
   Active: failed (Result: exit-code) since Wed 2018-10-24 10:33:49 CEST; 17min ago
  Process: 7200 ExecStart=/usr/bin/snap run multipass.multipassd (code=exited, status=1/FAILURE)
 Main PID: 7200 (code=exited, status=1/FAILURE)

paź 24 10:33:49 galeon systemd[1]: snap.multipass.multipassd.service: Service RestartSec=100ms expired, scheduling restart.
paź 24 10:33:49 galeon systemd[1]: snap.multipass.multipassd.service: Scheduled restart job, restart counter is at 5.
paź 24 10:33:49 galeon systemd[1]: Stopped Service for snap application multipass.multipassd.
paź 24 10:33:49 galeon systemd[1]: snap.multipass.multipassd.service: Start request repeated too quickly.
paź 24 10:33:49 galeon systemd[1]: snap.multipass.multipassd.service: Failed with result 'exit-code'.
paź 24 10:33:49 galeon systemd[1]: Failed to start Service for snap application multipass.multipassd.
```

The relevant piece of log is:
```
paź 24 10:08:16 galeon multipass.multipassd[4148]: [warning] [Qt] "No such interface “org.freedesktop.DBus.Properties” on object at path /org/freedesktop/NetworkManager/ActiveConnection/4"
paź 24 10:08:16 galeon multipass.multipassd[4148]: [error] [daemon] Could not determine group id for 'sudo'.
paź 24 10:08:16 galeon systemd[1]: snap.multipass.multipassd.service: Main process exited, code=exited, status=1/FAILURE
```

Apparently there is an assumption that sudo group must be present in the system: https://github.com/CanonicalLtd/multipass/blob/6b83f4073a45b19ca5594f5524e98ebe2434d507/src/daemon/daemon_main.cpp#L52-L72 which may not always be the case. For instance Arch does not have a sudo group. I checked with Fedora 28 cloud image, same thing there, no sudo group. Given that `multipass` is used by `snapcraft` to enabled building snaps on non Ubuntu systems, this may cause problems down the road.

Perhaps it would be best to leave the gorup ownership as `root:root` when there's no sudo available. Alternatively, since it's a classic snap, maybe you could try to add a `multipass` group in the install hook and use that for ownership.

### LINKED ISSUE #460: Bad messaging when `multipass` can't connect to the socket
```
$ multipass find
find failed: Connect Failed
```

There may be multiple reasons why you wouldn't be able to connect to the socket - the daemon being dead, the socket being in a different place, or the permissions being wrong. All of that results in the above, not giving much of a clue on what's going on.

### PR DESCRIPTION
**Pull request by townsend2010**
I combined a few fixes in this as separate commits since they all went hand in hand and solve a couple of issues together.

Fixes #456, fixes #460

### CODE REVIEW

File: include/multipass/cli/command.h [hunk 1]
```diff
@@ -81,6 +86,24 @@ class Command
...
+            {
+                auto socket_path = tokens[1];
+                QLocalSocket multipassd_socket;
+                multipassd_socket.connectToServer(QString::fromStdString(socket_path));
+                if (!multipassd_socket.waitForConnected() &&
+                    multipassd_socket.error() == QLocalSocket::SocketAccessError)
+                {
+                    grpc::Status denied_status{
+                        grpc::StatusCode::PERMISSION_DENIED, "multipass socket accesss denied",
+                        fmt::format("Please check that your user has group permission to access '{}'", socket_path)};
```
Comment (Saviq) [inline include/multipass/cli/command.h]:
```suggestion
                        fmt::format("Please check that your user has read-write permission to '{}'", socket_path)};
```

File: include/multipass/cli/command.h [hunk 2]
```diff
@@ -81,6 +86,24 @@ class Command
         {
             return on_success(reply);
         }
+        else
+        {
+            const auto tokens = multipass::utils::split(context.peer(), ":");
+            if (tokens[0] == "unix")
```
Comment (Saviq) [inline include/multipass/cli/command.h]: Could we do the same for network sockets?

Comment (townsend2010) [reply include/multipass/cli/command.h]: Sure, it's possible, but I think that may be beyond the scope of this PR.  But if you want to wrap that check into this PR so we don't lose the task to do this, I'm fine with that too:)

**Comment by Saviq** [comment]
This should give me a better error now, no?

```
$ git log --oneline master..HEAD | cat
d4d8c6d Merge 2fadd5a91751bb9333815096ffff655b9b591799 into 59243da528631ae63586c6a3765e034a6e1c9d78
2fadd5a Reword text for telling user to check permissions on the socket
f928677 launch: Change error formatting strings based on last commit
5f4f702 client: Check if socket can be connected on rpc failure
5587747 daemon: Use the 'sudo' or 'adm' group for the socket

$ multipass version
multipass  2018.10.1-135-gd4d8c6d
$ snap info multipass
name:      multipass
summary:   Ubuntu at your fingertips
publisher: –
license:   unset
description: |
  Multipass gives you Ubuntu VMs in seconds. Just run `multipass launch`
  and it'll do all the setup for you. Check out `multipass find` for a list
  of available images. More details you can find under `multipass help`.
commands:
  - multipass
  - multipass.virsh
services:
  multipass.libvirt-bin: simple, enabled, inactive
  multipass.multipassd:  simple, enabled, inactive
tracking:     edge
refresh-date: today at 15:16 CET
channels:
  stable:    –
  candidate: –
  beta:      2018.10.1              (461) 104MB classic
  edge:      2018.10.1-130-g59243da (527) 109MB classic
installed:   2018.10.1-135-gd4d8c6d (x2)  18B   classic,try
$ multipass ls
list failed: Connect Failed
```

**Comment by townsend2010** [comment]
@Saviq,

Not if multipassd is not running, which seems to be your case based on the `multipass version` output. The error message will be better only if multipassd is running and you don't have permission to access the socket.

File: include/multipass/cli/command.h [hunk 3]
```diff
@@ -81,6 +86,24 @@ class Command
...
+            {
+                auto socket_path = tokens[1];
+                QLocalSocket multipassd_socket;
+                multipassd_socket.connectToServer(QString::fromStdString(socket_path));
+                if (!multipassd_socket.waitForConnected() &&
+                    multipassd_socket.error() == QLocalSocket::SocketAccessError)
```
[the remark this answers is not in the document]
Comment (townsend2010) [reply include/multipass/cli/command.h]: You get the `Connect Failed` error from gRPC you saw above.  gRPC errors are horrendous and unhelpful really.

I'd really like to only target the bug this PR is addressing.  For trying to circumvent gRPC's connection problem shortcomings, I'd really like that to be a separate task.

File: include/multipass/cli/command.h [hunk 4]
```diff
@@ -81,6 +86,24 @@ class Command
...
+                {
+                    grpc::Status denied_status{
+                        grpc::StatusCode::PERMISSION_DENIED, "multipass socket access denied",
+                        fmt::format("Please check that your user has read-write permission to '{}'", socket_path)};
```
Comment (Saviq) [inline include/multipass/cli/command.h]:
```suggestion
                        fmt::format("Please check that you have read/write permissions to '{}'", socket_path)};
```

File: include/multipass/cli/command.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (Saviq) [reply include/multipass/cli/command.h]: Well, yeah, but couldn't we just print the message I mentioned in an `else` below?

[the remark this answers is not in the document]
Comment (townsend2010) [reply include/multipass/cli/command.h]: Ok, I think I understand your point now:)  I'll put something in and you can see if you like the wording.

[the remark this answers is not in the document]
Comment (townsend2010) [reply include/multipass/cli/command.h]: Done

File: include/multipass/cli/command.h [hunk 5]
```diff
@@ -81,6 +86,31 @@ class Command
...
+                    grpc::Status denied_status{
+                        grpc::StatusCode::PERMISSION_DENIED, "multipass socket access denied",
+                        fmt::format("Please check that you have read/write permissions to '{}'", socket_path)};
+                    return on_failure(denied_status);
+                }
+                else
+                {
+                    grpc::Status access_error_status{
+                        grpc::StatusCode::NOT_FOUND, "cannot connect to the multipass socket",
+                        fmt::format("Please ensure multipassd is running and '{}' exists", socket_path)};
```
[the remark this answers is not in the document]
Comment (Saviq) [reply include/multipass/cli/command.h]: Can we also put it one level up so it applies to TCP sockets, too?

```
$ MULTIPASS_SERVER_ADDRESS=127.0.0.2:50051 multipass ls
list failed: Connect Failed
```

[the remark this answers is not in the document]
Comment (townsend2010) [reply include/multipass/cli/command.h]: Done

File: include/multipass/cli/command.h [hunk 6]
```diff
@@ -81,8 +86,32 @@ class Command
         {
             return on_success(reply);
         }
-
-        return on_failure(status);
+        else
+        {
+            fmt::print("{}\n", context.peer());
```
Comment (Saviq) [inline include/multipass/cli/command.h]: Debug leftover?

**Comment by Saviq** [comment]
bors delegate+

bors?

File: include/multipass/cli/command.h (hunk 6 above)
Comment (townsend2010) [reply include/multipass/cli/command.h]: Oops:) Done

**Comment by townsend2010** [comment]
bors r=Saviq

**Comment by townsend2010** [comment]
One more time...

bors r=Saviq

**Pull request merged**